yesod-bin/Scaffolding/Scaffolder.hs

```diff
@@ -19,15 +20,18 @@ import           Data.Maybe            (isJust)
 import           Distribution.Text     (simpleParse)
 import           Distribution.Package  (PackageName)
 
-prompt :: (String -> Maybe a) -> IO a
-prompt f = do
-    s <- getLine
-    case f s of
+prompt :: (String -> Maybe a) -> Maybe String -> IO a
+prompt f (Just s) = accept f s
```

Comment:
It seems strange that, if the command line input is invalid, we'd switch to interactive mode. It would seem more logical to just error out in that case.